### DISCUSSION TITLE
更换 auth provider 的话，会影响聊天记录吗？

### DISCUSSION THREAD

**Discussion by sqkkyzx** [lobehub/lobehub#4285] [General Question]
想要请教一下大家，我现在希望更换 auth provider ，但是不想影响到以前的聊天记录，有人试过吗？有没有方案参考？

1. 如果从一个 auth provider 更换到另一个 auth provider ，它依赖什么来标识用户？用户名还是什么？
2. 如果直接更换行不通，那我直接更改数据库可以吗？如果更改数据库，直接修改新生成的用户的id为以前的id可以吗？还是需要将所有与原来用户id关联的记录指向新的id？

**Comment by BiFangKNT** [comment]
> 1. 如果从一个 auth provider 更换到另一个 auth provider ，它依赖什么来标识用户？用户名还是什么？

目前是以邮箱为主键（似乎还有用户名`uid`）

> 2. 如果直接更换行不通，那我直接更改数据库可以吗？如果更改数据库，直接修改新生成的用户的id为以前的id可以吗？还是需要将所有与原来用户id关联的记录指向新的id？

如果你新旧SSO都是同一邮箱，那么可以直接更换，在lobe中会被视为同一账号。https://github.com/lobehub/lobe-chat/discussions/4232#discussioncomment-10804579
可以改数据库。改新账户主键应该无法达成期望效果，改了之后首先在登录这一步应该就会出错。

**Comment by SpeedupMaster** [comment]
把数据库里面原来的ID改成新的ID就可以了
